airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/demo/file_based/platform/RecordWriter.java

```diff
@@ -0,0 +1,27 @@
+package io.airbyte.integrations.destination.snowflake.demo.file_based.platform;
+
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
+
+/**
+ * Writes airbyte records to a local file (i.e. within the connector container).
+ * <p>
+ * I don't love this interface, it's mostly illustrative. The existing SerializedBuffer stuff is
+ * approximately equivalent - I just didn't want to wire up all the dependencies.
+ */
+public interface RecordWriter {
```

Comment:
this is mostly a reaction to me not at all understanding the SerializableBuffer classes, and not actually a proposal to replace them.

My mental model of what that set of classes can do is:
* you can push in some amount of records, which get written to a file
* at some point, you can rotate the file, at which point new records get written to a new file and the old file can be uploaded to the destination

which is what I'm trying to capture here.